lib/wallet/open_channel.dart

```diff
@@ -24,20 +24,25 @@ class OpenChannel extends StatefulWidget {
 }
 
 class _OpenChannelState extends State<OpenChannel> {
-  int takerChannelAmount = OpenChannel.maxChannelAmount;
+  late int takerChannelAmount;
   bool submitting = false;
 
   final _formKey = GlobalKey<FormState>();
   static const minValue = 1000;
 
   @override
   void initState() {
-    super.initState();
-  }
+    final bitcoinBalance = context.read<BitcoinBalance>();
 
-  @override
-  void dispose() {
-    super.dispose();
+    final totalBalance = bitcoinBalance.confirmed.asSats +
+        bitcoinBalance.pendingInternal.asSats +
+        bitcoinBalance.pendingExternal.asSats;
+
+    final maxTakerChannelAmount = (totalBalance).clamp(0, OpenChannel.maxChannelAmount);
```

Comment:
nit: you could directly set the `takerChannelAmount` here, instead of using `maxTakerChannelAmount`